### PR TITLE
[SUB-TASK][KPIP-4] Support internal rest request authentication to enable http request redirection across kyuubi instances

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Support internal rest request authentication to enable redirect http request across kyuubi instances

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala [hunk 1]
```diff
@@ -151,13 +151,21 @@ object HighAvailabilityConf {
       .booleanConf
       .createWithDefault(false)
 
-  val HA_ZK_ENGINE_SECURE_SECRET_NODE: OptionalConfigEntry[String] =
+  @deprecated(s"using ${HA_ZK_INTERNAL_SECURE_SECRET_NODE.key} instead, 1.6.0")
```
Comment (ulysses-you) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala:154]: ditto

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -1248,59 +1248,112 @@ object KyuubiConf {
...
+  @deprecated(s"using ${INTERNAL_SECURITY_CRYPTO_CIPHER_TRANSFORMATION.key} instead", "1.6.0")
   val ENGINE_SECURITY_CRYPTO_CIPHER_TRANSFORMATION: ConfigEntry[String] =
     buildConf("kyuubi.engine.security.crypto.cipher")
-      .doc("The cipher transformation to use for encrypting engine access token.")
+      .doc("(deprecated) The cipher transformation to use for encrypting engine access token.")
       .version("1.5.0")
       .stringConf
       .createWithDefault("AES/CBC/PKCS5PADDING")
 
+  val INTERNAL_SECURITY_ENABLED: ConfigEntry[Boolean] =
```
Comment (ulysses-you) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1313]: not sure such rename is necessary, a internal tag seems enough

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/http/authentication/KyuubiInternalAuthenticationHandler.scala [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      response: HttpServletResponse): String = {
+    var authUser: String = null
+    val authorization = getAuthorization(request)
+    val inputToken = Option(authorization).map(a => Base64.getDecoder.decode(a.getBytes()))
+      .getOrElse(Array.empty[Byte])
+    val creds = new String(inputToken, Charset.forName("UTF-8")).split(":")
+
+    if (creds.size < 2 || creds(0).trim.isEmpty || creds(1).trim.isEmpty) {
+      response.setHeader(WWW_AUTHENTICATE, authScheme.toString)
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED)
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/http/authentication/KyuubiInternalAuthenticationHandler.scala:55]: shall we add a test for this branch ?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 4]
```diff
@@ -1248,59 +1248,112 @@ object KyuubiConf {
       .stringConf
       .createWithDefault("engine_operation_logs")
 
+  @deprecated(s"using ${INTERNAL_SECURITY_ENABLED.key} instead", "1.6.0")
```
[the remark this answers is not in the document]
Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1251]: thanks for the suggestions

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/http/authentication/KyuubiInternalAuthenticationHandler.scala (hunk 3 above)
Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/http/authentication/KyuubiInternalAuthenticationHandler.scala:55]: sure

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: add more docs for new config item

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 2 above)
Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: add internal tag and complete the docs

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala [hunk 5]
```diff
@@ -151,11 +151,12 @@ object HighAvailabilityConf {
       .booleanConf
       .createWithDefault(false)
 
-  val HA_ZK_ENGINE_SECURE_SECRET_NODE: OptionalConfigEntry[String] =
-    buildConf("kyuubi.ha.zookeeper.engine.secure.secret.node")
-      .doc("The zk node contains the secret that used for internal secure between Kyuubi server " +
-        "and Kyuubi engine, please make sure that it is only visible for Kyuubi.")
-      .version("1.5.0")
+  val HA_ZK_INTERNAL_SECURE_SECRET_NODE: OptionalConfigEntry[String] =
+    buildConf("kyuubi.ha.zookeeper.internal.secure.secret.node")
```
Comment (ulysses-you) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala:155]: revert the change

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala [hunk 6]
```diff
@@ -153,9 +153,10 @@ object HighAvailabilityConf {
 
   val HA_ZK_ENGINE_SECURE_SECRET_NODE: OptionalConfigEntry[String] =
     buildConf("kyuubi.ha.zookeeper.engine.secure.secret.node")
-      .doc("The zk node contains the secret that used for internal secure between Kyuubi server " +
-        "and Kyuubi engine, please make sure that it is only visible for Kyuubi.")
-      .version("1.5.0")
+      .internal
+      .doc("The zk node contains the secret that used for internal secure, please make sure " +
+        "that it is only visible for Kyuubi.")
+      .version("1.6.0")
```
Comment (ulysses-you) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala:159]: nit: 1.5.0

**Review by ulysses-you — APPROVED** [review]

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala (hunk 6 above)
Comment (turboFei) [reply kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala]: thanks

**Comment by turboFei** [comment]
thanks, merged to master
